homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,305 @@
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.data = user_input
+            return await self.async_step_options()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_SETUP,
+            errors=errors,
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle remaining flow."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.data, **user_input}
+            if combined_input[CONF_PROVINCE] == "none":
+                combined_input[CONF_PROVINCE] = None
+
+            try:
+                await self.hass.async_add_executor_job(
+                    validate_custom_dates, combined_input
+                )
+            except AddDatesError:
+                errors["add_holidays"] = "add_holiday_error"
+            except RemoveDatesError:
+                errors["remove_holidays"] = "remove_holiday_error"
+            except NotImplementedError:
+                self.async_abort(reason="incorrect_province")
+
+            abort_match = {
+                CONF_COUNTRY: combined_input[CONF_COUNTRY],
+                CONF_EXCLUDES: combined_input[CONF_EXCLUDES],
+                CONF_OFFSET: combined_input[CONF_OFFSET],
+                CONF_WORKDAYS: combined_input[CONF_WORKDAYS],
+                CONF_ADD_HOLIDAYS: combined_input[CONF_ADD_HOLIDAYS],
+                CONF_REMOVE_HOLIDAYS: combined_input[CONF_REMOVE_HOLIDAYS],
+                CONF_PROVINCE: combined_input[CONF_PROVINCE],
+            }
+
+            self._async_abort_entries_match(abort_match)
+            if not errors:
+                return self.async_create_entry(
+                    title=combined_input[CONF_NAME],
+                    data={},
+                    options=combined_input,
+                )
+
+        schema = await self.hass.async_add_executor_job(
+            add_province_to_schema, DATA_SCHEMA_OPT, self.data
+        )
+        new_schema = self.add_suggested_values_to_schema(schema, user_input)
+        return self.async_show_form(
+            step_id="options",
+            data_schema=new_schema,
+            errors=errors,
+        )
+
+
+class WorkdayOptionsFlowHandler(OptionsFlowWithConfigEntry):
+    """Handle Workday options."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage Workday options."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.options, **user_input}
+            if combined_input[CONF_PROVINCE] == "none":
```

Comment:
It might be possible to hide the "province" if provinces are not available and use:
```suggestion
            if combined_input.get(CONF_PROVINCE, NONE_SENTINEL) == NONE_SENTINEL:
```